File: ROS1_src/learning_parameter/src/parameter_config.cpp
//该例程设置/读取海龟例程中的参数
#include<string>
#include<ros/ros.h>
#include<std_srvs/Empty.h>

int main(int argc,char**argv)
{
    int red,green,blue;
    ros::init(argc,argv,"parameter_config");
    ros::NodeHandle node;

    //读取背景颜色参数
    ros::param::get("/turtlesim/background_r",red);
    ros::param::get("/turtlesim/background_g",green);
    ros::param::get("/turtlesim/background_b",blue);

    ROS_INFO("Get Background Color[%d,%d,%d]",red,green,blue);

    //设置背景颜色参数
    ros::param::set("/turtlesim/background_r",255);
    ros::param::set("/turtlesim/background_g",255);
    ros::param::set("/turtlesim/background_b",255);
    ROS_INFO("Set Background Color[255,255,255]");

    //读取背景颜色参数,可以省略的一步
    ros::param::get("/turtlesim/background_r",red);
    ros::param::get("/turtlesim/background_g",green);
    ros::param::get("/turtlesim/background_b",blue);
    ROS_INFO("Re-get Background Color[%d,%d,%d]",red,green,blue);

    //调用服务，刷新背景颜色
    ros::service::waitForService("/clear");
    ros::ServiceClient clear_background=node.serviceClient<std_srvs::Empty>("/clear");
    std_srvs::Empty srv;
    clear_background.call(srv);

    sleep(1);
    return 0;
}

File: ROS1_src/learning_parameter/scripts/parameter_config.py
#！/usr/bin/env python
#-*-coding:utf-8-*-
import sys
import rospy
from std_srvs.srv import Empty

def parameter_config():
    rospy.init_node('parameter_config',anonymous=True)

    red=rospy.get_param('/turtlesim/background_r')
    green=rospy.get_param('/turtlesim/background_g')
    blue=rospy.get_param('/turtlesim/background_b')
    rospy.loginfo("Get Background color[%d,%d,%d]",red,green,blue)

    rospy.set_param('/turtlesim/background_r',255)
    rospy.set_param('/turtlesim/background_g',255)
    rospy.set_param('/turtlesim/background_b',255)
    rospy.loginfo("Set Background Color[255,255,255]")

    red=rospy.get_param('/turtlesim/background_r')
    green=rospy.get_param('/turtlesin/background_g')
    blue=rospy.get_param('/turtlesin/background_b')
    rospy.loginfo("Get Background color[%d,%d,%d]",red,green,blue)

    #发现/clear服务后，创建一个服务客户端，连接名为/clear的service
    rospy.wait_for_service('/clear')
    try:
        clear_background=rospy.ServiceProxy('/clear',Empty)
        
        #请求服务调用，输入请求数据
        response=clear_background()
        return response
    except rospy.ServiceException,e:
        print "Service Call failed:%s"%e

if __name__=='__main__':
    parameter_config()